modules/gateway/upnp.go

```diff
@@ -77,6 +76,8 @@ func (g *Gateway) threadedLearnHostname() {
 		return
 	}
 
+	// TODO: Is this safe? What happens to processes or nodes that have already
+	// request the IP address?
```

Comment:
Interestingly the only place we use `myAddr` in the gateway is in `addNode` to ensure we don't add ourself as a node. Perhaps `gateway.Address()` should return an error or bool if the hostname hasn't been discovered yet.

But to get to the point, this is currently safe in the current code, but it may be necessary to check `myAddr.IsValid()` before using `myAddr` in future code.